File: pubnub-gson/src/main/java/com/pubnub/api/endpoints/files/ListFiles.java
package com.pubnub.api.endpoints.files;

import com.pubnub.api.endpoints.BuilderSteps;
import com.pubnub.api.endpoints.Endpoint;
import com.pubnub.api.models.consumer.files.PNListFilesResult;

public interface ListFiles extends Endpoint<PNListFilesResult> {
    ListFiles limit(Integer limit);

    ListFiles next(com.pubnub.api.models.consumer.objects.PNPage.PNNext next);

    interface Builder extends BuilderSteps.ChannelStep<ListFiles> { }
}

File: pubnub-kotlin/pubnub-kotlin-api/src/commonMain/kotlin/com/pubnub/kmp/endpoints/files/ListFiles.kt
package com.pubnub.kmp.endpoints.files

import com.pubnub.kmp.PNFuture
import com.pubnub.api.models.consumer.files.PNListFilesResult

/**
 * @see [PubNub.listFiles]
 */
interface ListFiles : PNFuture<PNListFilesResult>